### PR TITLE
appdev: an assortment of ux updates

### PR DESCRIPTION
**Pull request by kamaln7**
the diff is 99% vendor changes i promise 😛 

* highlight current app dev workspace in multiple places
* style fixups
* print next steps after a successful build
* use correct image names for static sites
* improved onboarding
* functions build guidance
* print custom builder image override to screen
* fix image pull progress bar

### CODE REVIEW

File: commands/apps_dev.go [hunk 1]
```diff
@@ -188,6 +197,17 @@ func RunAppsDevBuild(c *CmdConfig) error {
 	if !ok {
 		return fmt.Errorf("cannot build component %s", componentName)
 	}
+
+	if componentSpec.GetType() == godo.AppComponentTypeFunctions {
+		template.Print(heredoc.Doc(`
+
+			{{warning (print crossmark " functions builds are coming soon!")}}
+			  please use {{highlight "doctl serverless sls deploy"}} to build functions in the meantime.
```
Comment (andrewsomething) [inline commands/apps_dev.go:205]:
```suggestion
			  please use {{highlight "doctl serverless deploy"}} to build functions in the meantime.
```

File: commands/apps_dev.go [hunk 2]
```diff
@@ -375,10 +402,21 @@ func RunAppsDevBuild(c *CmdConfig) error {
...
+			heredoc.Doc(`
+				{{success checkmark}} successfully built {{success .component}} in {{highlight (duration .dur)}}
+				{{success checkmark}} created container image {{success .img}}
+				
+				{{pointerRight}} push your image to a container registry using {{highlight "docker push"}}
+				{{pointerRight}} or run it locally using {{highlight "docker run"}}; for example:
+				  
+				   {{muted promptPrefix}} {{highlight (printf "docker run --rm -p 8080:8080 %s" .img)}}
+				
+				  then access your component at {{underline "http://localhost:8080"}}`,
```
Comment (andrewsomething) [inline commands/apps_dev.go:414]: :ok_hand: This is a good addition. Was going to mention that the post-build experience left a lot of the user to figure out on their own.

**Review by nicktate — DISMISSED** [review]
This looks great 🥳

**Review by nicktate — APPROVED** [review]

File: commands/apps_dev.go [hunk 3]
```diff
@@ -269,8 +293,11 @@ func RunAppsDevBuild(c *CmdConfig) error {
 	// }
 
 	buildingComponentLine := template.String(
-		`building {{lower (snakeToTitle .GetType)}} {{highlight .GetName}}`,
-		componentSpec,
+		`building {{lower (snakeToTitle .componentSpec.GetType)}} {{highlight .componentSpec.GetName}} {{muted print "(" .appName ")"}}`,
```
Comment (nicktate) [inline commands/apps_dev.go:296]: muted takes only 1 arg and this will cause a failure

**Review by nicktate — CHANGES_REQUESTED** [review]

**Review by nicktate — APPROVED** [review]

File: commands/apps_dev.go (hunk 3 above)
Comment (kamaln7) [reply commands/apps_dev.go]: fixed

**Pull request merged**